ReplayMessenger.extended.lib.nut

```diff
@@ -0,0 +1,562 @@
+        local msg = Message(_getNextId(), name, data, importance, ackTimeout, metadata);
+        _send(msg);
+        return msg;
+    }
+
+    /**
+    * The handler to be called when ReplayMessenger is replaying a persisted message.
+    *
+    * @callback confirmResendCallback
+    * @param {Message} msg - The message (an instance of class Message) being replayed
+    *
+    * @return {boolean} - `true` to confirm message resending or `false` to drop the message
+    */
+    /**
+    * Sets the handler to be called when ReplayMessenger is replaying a persisted message.
+    *
+    * @param {confirmResendCallback} confirmResendCallback - The handler.
+    *   It should return `true` to confirm message resending or `false` to drop the message.
+    */
+    function confirmResend(confirmResendCallback) {
+        _confirmResend = confirmResendCallback;
+    }
+
+    // -------------------- PRIVATE METHODS -------------------- //
+
+    // Sends the message (and immediately persists it if needed) and restarts the timer for processing the queues
+    function _send(msg) {
+        // Check if the message has importance = RM_IMPORTANCE_CRITICAL and not yet persisted
+        if(msg._importance == RM_IMPORTANCE_CRITICAL && !_isMsgPersisted(msg)) {
+            _persistMessage(msg);
+        }
+
+        base._send(msg);
+    }
+
+    // The handler for message acknowledgement
+    function _onAckReceived(payload) {
+        local id = payload.id;
+
+        _log("Ack received. Msg id: " + id);
+
+        if (id in _sentQueue) {
+            local msg = _sentQueue[id];
+
+            // If the message is persisted, erase it from the flash
+            if(_isMsgPersisted(msg)) {
+                _safeEraseMsg(id, msg);
+            } else if (msg._importance >= RM_IMPORTANCE_HIGH) {
+                // The message can also be in _persistMessagesQueue.
+                // If so, we should remove it from there
+                foreach (idx, msgToPersist in _persistMessagesQueue) {
+                    if (msgToPersist.payload.id == id) {
+                        _persistMessagesQueue.remove(idx);
+                    }
+                }
+            }
+
+            delete _sentQueue[id];
+
+            // Check if there is some data sent along with the acknowledgement
+            local data = null;
+            if("data" in payload) {
+                data = payload.data;
+            }
+
+            // Call the handler if set
+            if (_isFunction(_onAck)) {
+                _onAck(msg, data);
+            }
+        }
+    }
+
+    // Calls the error handler if set for the message with the specified error
+    function _onSendFail(msg, error) {
+        local id = msg.payload.id;
+        _log("Failed to send or haven't received Ack. Id: " + id + " Reason: " + error);
+
+        // If importance is RM_IMPORTANCE_HIGH, persist the message if not yet
+        if(msg._importance == RM_IMPORTANCE_HIGH && !_isMsgPersisted(msg)) {
```

Comment:
Can we please make the code conformant with the coding style requirements.
"if("->"if (", etc.

Here and everywhere, I won't post it for every single case I find.

Reply:
Fixed!